modules/apps/27-interchain-accounts/types/metadata.go

```diff
@@ -120,13 +120,15 @@ func ValidateHostMetadata(ctx sdk.Context, channelKeeper ChannelKeeper, connecti
 		return errorsmod.Wrapf(ErrUnknownDataType, "unsupported transaction type %s", metadata.TxType)
 	}
 
-	connection, err := channelKeeper.GetConnection(ctx, connectionHops[0])
-	if err != nil {
-		return err
-	}
+	if len(connectionHops) > 0 {
```

Comment:
`len(connectionHops)` should never zero in this case. 

Any reason this PR is changing `ValidateHostMetadata` when its not being used here? 

This would also make this implementation different form `ValidateControllerMetadata` above - I'd imagine they both should be consistent in what they're doing.

Reply:
i'll revert for now -- can also simplify this file to have one `ValidateMetadata` function if they're doing the same thing then

Reply:
they're not doing exactly the same thing, they pass different args to `validateConnectionParams` as the self/counterparty values will be different/opposing when called from either controller or host

Reply:
we still need both functions at the moment unless a bigger refactor was done to accommodate it differently somehow, but I think we will always need to handle this one way or another

Reply:
yes, i know, but can't we just add a boolean flag to the Validate function to indicate if the caller is a controller module or host?

Reply:
i'll push the code in a sec so you can see what it looks like, if you don't like it we can revert and keep both functions.